main/src/ca/uwaterloo/flix/language/phase/unification/EffUnification3.scala

```diff
@@ -130,22 +145,28 @@ object EffUnification3 {
     }
   }
 
+  /**
+   * Returns the given list of type equations as a list of set equations.
+   */
+  private def toEquations(l: List[(Type, Type, SourceLocation)], withSlack: Boolean)(implicit scope: Scope, renv: RigidityEnv, m: SortedBimap[Atom, Int]): List[Equation] =
+    l.map(e => toEquation(e, withSlack))
+
   /**
     * Translates `eq` into a equation of [[SetFormula]].
     *
     * Throws [[InvalidType]] for types not convertible to [[SetFormula]].
     */
-  private def toEquation(eq: (Type, Type, SourceLocation))(implicit scope: Scope, renv: RigidityEnv, m: SortedBimap[Atom, Int]): Equation = {
+  private def toEquation(eq: (Type, Type, SourceLocation), withSlack: Boolean)(implicit scope: Scope, renv: RigidityEnv, m: SortedBimap[Atom, Int]): Equation = {
     val (tpe1, tpe2, loc) = eq
-    Equation.mk(toSetFormula(tpe1), toSetFormula(tpe2), loc)
+    Equation.mk(toSetFormula(tpe1)(withSlack = withSlack, scope, renv, m), toSetFormula(tpe2)(withSlack = withSlack, scope, renv, m), loc)
   }
 
   /**
     * Returns `t` as a [[SetFormula]].
     *
     * Throws [[InvalidType]] if `t` is not valid.
     */
-  private def toSetFormula(t: Type)(implicit scope: Scope, renv: RigidityEnv, m: SortedBimap[Atom, Int]): SetFormula = t match {
+  private def toSetFormula(t: Type)(implicit withSlack: Boolean, scope: Scope, renv: RigidityEnv, m: SortedBimap[Atom, Int]): SetFormula = t match {
```

Comment:
Implicit Boolean seems evil

Reply:
Agreed. But there is a ticket to introduce something better.